### ISSUE TITLE
Implement C-style for loops

### ISSUE THREAD

**Issue by puppeh** [oilshell/oil#66]
This appears to be a missing feature:

```bash
#!/bin/bash
for ((i=0; i<10; i++)); do
  echo $i
done
```

Result with osh 0.3.0:

```
Traceback (most recent call last):
  File "/home/andy/git/oilshell/oil/Python-2.7.13/Lib/runpy.py", line 174, in _run_module_as_main
  File "/home/andy/git/oilshell/oil/Python-2.7.13/Lib/runpy.py", line 72, in _run_code
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 509, in <module>
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 492, in main
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 475, in OilMain
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 413, in OshMain
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 985, in Execute
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 949, in _Execute
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 758, in _Dispatch
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 978, in _ExecuteList
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 949, in _Execute
  File "/home/andy/git/oilshell/oil/bin/../core/cmd_exec.py", line 845, in _Dispatch
NotImplementedError: 14
```

Result with bash 4.3.42:

```
0
1
2
3
4
5
6
7
8
9
```

**Comment by andychu** [comment]
I just published 0.4.alpha1 and it has this feature:

http://www.oilshell.org/releases.html

Let me know if it doesn't work!

**Comment by andychu** [comment]
Fixed in 0.4.0 http://www.oilshell.org/blog/2018/02/03.html
